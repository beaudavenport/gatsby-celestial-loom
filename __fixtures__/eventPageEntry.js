export default {
  markdownRemark: {
    html: '<p>Join us A HUNDRED YEARS FROM NOW for a Goddess GEMINI Moon (12 Gemini) celebration at our sacred MOON CIRCLE.  We will explore the making "connections" lunar unfolding in June, learn about the "Lights Mandala" and apply our personal chart Lunar Planner to our monthly timing.</p>',
    frontmatter: {
      eventDate: 'May 30, 2119',
      eventTime: '7 PM - 9 PM',
      title: 'Moon Circle of the Future',
      subtitle: 'Gemini NEW MOON Celebration',
      image: '/assets/gemini-moon-circle.jpg',
      location: 'Divine Inspirations',
      mapsLink: 'https://goo.gl/maps/MCzF4v3UguEXhmeq6',
      eventPrice: 25,
      priceDescription: '2 Hour Playshop',

    },
    fields: { slug: '/events/moon-circle/' },
  },
};
