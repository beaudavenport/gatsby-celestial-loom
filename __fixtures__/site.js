export default {
  site: {
    siteMetadata: {
      title: 'The Celestial Loom',
    },
  },
};
