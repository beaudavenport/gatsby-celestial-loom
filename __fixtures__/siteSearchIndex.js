export default {
  siteSearchIndex: {
    index: null,
  },
};
