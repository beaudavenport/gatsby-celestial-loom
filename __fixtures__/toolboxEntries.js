export default {
  allMarkdownRemark: {
    edges: [
      {
        node: {
          id: '58aa600c-e761-5ebc-b285-eb6d59ecc4ac',
          frontmatter: { title: 'Aries', type: 'toolbox', toolboxType: 'Signs' },
          excerpt: 'Key Phrase:  I AM Basic Nature:  Pioneer and Warrior Keywords:  courageous, independent, assertive, spontaneous, head of cosmic man, self…',
          fields: { slug: '/toolbox/aries/' },
        },
      }, {
        node: {
          id: 'e9866973-380c-54ee-96c4-6a44b09dc958',
          frontmatter: { title: 'First House', type: 'toolbox', toolboxType: 'Houses' },
          excerpt: 'I AM -  The appearance, the basic mood in life, personality, vital energy, awareness of life, physical constitution, vitality, expression of…',
          fields: { slug: '/toolbox/first-house/' },
        },
      }, {
        node: {
          id: 'c96e763f-d670-5bef-a05c-3792d86a0a60',
          frontmatter: { title: 'Sun', type: 'toolbox', toolboxType: 'Planets' },
          excerpt: 'Everything in the Solar System revolves around the Sun. Growth, self-expression, outer-world activities Positive energy in any endeavor Tap…',
          fields: { slug: '/toolbox/sun/' },
        },
      }, {
        node: {
          id: 'f1fc71be-24db-5613-991a-0fdc62d5e03b',
          frontmatter: { title: 'Moon', type: 'toolbox', toolboxType: 'Planets' },
          excerpt: 'In our Solar System, the Moon reflects the light. Fundamental energy tied to emotions and intuition Celestial power tied to the Goddess Moon…',
          fields: { slug: '/toolbox/moon/' },
        },
      }, {
        node: {
          id: 'a8eabb89-ed90-5b1b-a942-80aeee38994f',
          frontmatter: { title: 'Second House', type: 'toolbox', toolboxType: 'Houses' },
          excerpt: 'I HAVE  - Assets, foundation of material, intellectual, and emotional security, wealth, possessions, finances, social background, money…',
          fields: { slug: '/toolbox/second-house/' },
        },
      }, {
        node: {
          id: '693f6b38-1f10-5555-b6f8-43cbaac2d26e',
          frontmatter: { title: 'Taurus', type: 'toolbox', toolboxType: 'Signs' },
          excerpt: 'Key Phrase:  I HAVE Basic Nature:  Builder and Accumulator Keywords:  courageous, independent, assertive, spontaneous, head of cosmic man…',
          fields: { slug: '/toolbox/taurus/' },
        },
      }, {
        node: {
          id: 'bbd334c8-9405-5f26-98d9-9617ebf82874',
          frontmatter: { title: 'Gemini', type: 'toolbox', toolboxType: 'Signs' },
          excerpt: 'Key Phrase:  I THINK Basic Nature:  Communicator and Threshold Walker Keywords:  intellectual, communicative, curious, versatile, lower neck…',
          fields: { slug: '/toolbox/gemini/' },
        },
      }, {
        node: {
          id: '80acb630-2187-5267-a599-c4359b0d68a3',
          frontmatter: { title: 'Mercury', type: 'toolbox', toolboxType: 'Planets' },
          excerpt: 'Mercury is the messenger of the Gods. Rules speech and thinking Tap energy when you use affirmations, sigils, hand gestures, and thought…',
          fields: { slug: '/toolbox/mercury/' },
        },
      },
    ],
  },
};
