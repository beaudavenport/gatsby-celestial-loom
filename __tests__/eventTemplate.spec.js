import { StaticQuery, useStaticQuery } from 'gatsby';
import React from 'react';

import renderer from 'react-test-renderer';

import EventTemplate from '../src/templates/eventTemplate';
import eventArchiveEntries from '../__fixtures__/eventArchiveEntries';
import eventPageEntry from '../__fixtures__/eventPageEntry';
import sidebarEntries from '../__fixtures__/sidebarEntries';
import site from '../__fixtures__/site';
import siteSearchIndex from '../__fixtures__/siteSearchIndex';
import toolboxEntries from '../__fixtures__/toolboxEntries';

beforeEach(() => {
  global.window.matchMedia = jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(),
    removeListener: jest.fn(),
  }));

  StaticQuery.mockImplementation(({ render }) => render({
    ...site,
    ...siteSearchIndex,
    ...eventArchiveEntries,
    ...sidebarEntries,
  }));

  useStaticQuery.mockReturnValueOnce(toolboxEntries);
});

describe('Event Template', () => {
  it('should render correctly with events archive, event page', () => {
    const tree = renderer
      .create(<EventTemplate data={eventPageEntry} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
