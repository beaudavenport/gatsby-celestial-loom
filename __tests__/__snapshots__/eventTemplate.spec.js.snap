// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Template should render correctly with events archive, event page 1`] = `
<div>
  <header
    className="md-paper md-paper--2 md-toolbar md-background--primary md-toolbar--text-white md-toolbar--fixed"
    style={
      Object {
        "backgroundColor": "rgb(47, 47, 47)",
      }
    }
  >
    <button
      className="md-btn md-btn--icon md-pointer--hover md-inline-block md-btn--toolbar md-toolbar--action-left"
      disabled={false}
      onClick={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="md-ink-container"
      />
      <i
        className="md-icon material-icons md-text--inherit"
      >
        menu
      </i>
    </button>
    <h2
      className="md-title md-title--toolbar"
      style={
        Object {
          "fontFamily": "Berkshire Swash",
          "fontStyle": "italic",
        }
      }
    >
      Events
    </h2>
    <div
      className="md-cell--right md-toolbar--action-right"
    >
      <div
        className="md-btn--toolbar header-icons-container"
      >
        <div
          className="header-search-container"
        >
          <button
            className="md-btn md-pointer--hover md-text md-inline-block"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
            type="button"
          >
            <div
              className="md-ink-container"
            />
            <div
              className="md-badge-container md-inline-block"
            >
              <i
                className="md-icon material-icons"
                style={
                  Object {
                    "fontSize": "2rem",
                  }
                }
              >
                shopping_cart
              </i>
            </div>
          </button>
        </div>
        <div
          className="header-search-container"
        >
          <button
            className="md-btn md-pointer--hover md-text md-inline-block"
            onClick={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
              }
            }
            type="button"
          >
            <div
              className="md-ink-container"
            />
            <i
              className="md-icon material-icons"
              style={
                Object {
                  "fontSize": "2rem",
                }
              }
            >
              search
            </i>
          </button>
        </div>
      </div>
    </div>
  </header>
  <main
    className="md-navigation-drawer-content md-navigation-drawer-content--inactive md-transition--acceleration md-toolbar-relative"
    id="main-content"
    tabIndex={-1}
  >
    <div
      className="md-grid md-grid--no-spacing"
      style={
        Object {
          "maxWidth": 1100,
        }
      }
    >
      <div
        className="md-cell md-cell--8 main-contents"
      >
        <div
          className="link--container false"
        >
          <a
            href="/events"
            style={
              Object {
                "textDecoration": "none",
              }
            }
          >
            <div
              className="touchable-link "
              style={Object {}}
            >
              <div
                className="md-ink-container"
              />
              <div
                className="flex-center"
              >
                <i
                  className="md-icon material-icons"
                  style={
                    Object {
                      "color": "#ec6602",
                      "fontSize": "2rem",
                      "marginRight": 5,
                    }
                  }
                >
                  arrow_back
                </i>
                <p
                  className="back-title"
                >
                  All Events
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <div
            style={
              Object {
                "marginBottom": 30,
                "textAlign": "center",
              }
            }
          >
            <h4
              className="subtitle"
            >
              Past Event
            </h4>
            <h2
              className="title"
            >
              Moon Circle
            </h2>
            <h4
              className="subtitle"
            >
              Gemini NEW MOON Celebration
            </h4>
          </div>
          <div
            style={
              Object {
                "margin": "0 auto",
                "maxWidth": 400,
              }
            }
          >
            <img
              alt="event-page"
              src="/assets/gemini-moon-circle.jpg"
              style={
                Object {
                  "margin": "0 auto",
                  "objectFit": "cover",
                  "width": "100%",
                }
              }
            />
          </div>
          <div
            style={
              Object {
                "margin": "0 auto",
                "maxWidth": "500px",
              }
            }
          >
            <div
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <div
                className="event-detail"
              >
                <i
                  className="md-icon material-icons"
                  style={
                    Object {
                      "marginRight": 5,
                    }
                  }
                >
                  place
                </i>
                <a
                  href="https://goo.gl/maps/MCzF4v3UguEXhmeq6"
                >
                  Divine Inspirations
                </a>
              </div>
              <div
                className="event-detail"
              >
                <i
                  className="md-icon material-icons"
                  style={
                    Object {
                      "marginRight": 5,
                    }
                  }
                >
                  event
                </i>
                <p
                  style={
                    Object {
                      "fontSize": "1.1rem",
                    }
                  }
                >
                  May 30, 2019 â€¢ 7 PM - 9 PM
                </p>
              </div>
            </div>
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>Join us for a Goddess GEMINI Moon (12 Gemini) celebration at our sacred MOON CIRCLE.  We will explore the making \\"connections\\" lunar unfolding in June, learn about the \\"Lights Mandala\\" and apply our personal chart Lunar Planner to our monthly timing.</p>",
                }
              }
              style={
                Object {
                  "border": "1px solid rgba(15,70,100,.12)",
                  "borderRadius": 5,
                  "padding": 20,
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        className="md-cell md-cell--4 md-cell--8-tablet sidebar-contents"
      >
        <div
          style={
            Object {
              "marginBottom": 30,
              "marginTop": 20,
            }
          }
        >
          <div
            className="sidebar-header"
          >
            <h5
              className="sidebar-header--title"
            >
              Events Calendar
            </h5>
          </div>
          <div>
            <ul
              className="md-list"
            >
              <div>
                <li
                  className="md-list-item md-list-item--nested-container"
                  style={Object {}}
                >
                  <div
                    aria-expanded={false}
                    aria-pressed={false}
                    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-list-tile md-list-tile--two-lines md-text"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      className="md-ink-container"
                    />
                    <div
                      className="md-tile-content"
                    >
                      <div
                        className="md-tile-text--primary md-text"
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        May 2019
                      </div>
                      <div
                        className="md-tile-text--secondary md-text--secondary"
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        1 events
                      </div>
                    </div>
                    <div
                      className="md-tile-addon md-tile-addon--icon"
                    >
                      <i
                        className="md-icon material-icons md-collapser"
                      >
                        keyboard_arrow_down
                      </i>
                    </div>
                  </div>
                </li>
              </div>
              <div>
                <li
                  className="md-list-item md-list-item--nested-container"
                  style={Object {}}
                >
                  <div
                    aria-expanded={false}
                    aria-pressed={false}
                    className="md-fake-btn md-pointer--hover md-fake-btn--no-outline md-list-tile md-list-tile--two-lines md-text"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <div
                      className="md-ink-container"
                    />
                    <div
                      className="md-tile-content"
                    >
                      <div
                        className="md-tile-text--primary md-text"
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        June 2019
                      </div>
                      <div
                        className="md-tile-text--secondary md-text--secondary"
                        style={
                          Object {
                            "fontWeight": "bold",
                          }
                        }
                      >
                        1 events
                      </div>
                    </div>
                    <div
                      className="md-tile-addon md-tile-addon--icon"
                    >
                      <i
                        className="md-icon material-icons md-collapser"
                      >
                        keyboard_arrow_down
                      </i>
                    </div>
                  </div>
                </li>
              </div>
            </ul>
          </div>
        </div>
        <div>
          <div
            className="sidebar-header"
          >
            <h5
              className="sidebar-header--title"
            >
              Latest Updates
            </h5>
          </div>
          <div
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <a
              href="/blog/norwac-i-am-on-my-way/"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="touchable-link thumbnail-card"
                style={Object {}}
              >
                <div
                  className="md-ink-container"
                />
                <div
                  className="thumbnail-image"
                  style={
                    Object {
                      "backgroundImage": "url(/assets/cancer-norwac.jpg)",
                    }
                  }
                />
                <div
                  className="thumbnail-text"
                >
                  <p
                    className="thumbnail-text--title"
                  >
                    NORWAC - I AM ON MY WAY
                  </p>
                  <p
                    className="thumbnail-text--subtitle"
                  >
                    Posted on:
                     
                    <span
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      May 23, 2019
                    </span>
                  </p>
                </div>
              </div>
            </a>
            <a
              href="/blog/the-pleiades/"
              style={
                Object {
                  "textDecoration": "none",
                }
              }
            >
              <div
                className="touchable-link thumbnail-card"
                style={Object {}}
              >
                <div
                  className="md-ink-container"
                />
                <div
                  className="thumbnail-image"
                  style={
                    Object {
                      "backgroundImage": "url(/assets/pleiades.jpg)",
                    }
                  }
                />
                <div
                  className="thumbnail-text"
                >
                  <p
                    className="thumbnail-text--title"
                  >
                    The Pleiades
                  </p>
                  <p
                    className="thumbnail-text--subtitle"
                  >
                    Posted on:
                     
                    <span
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontWeight": "bold",
                        }
                      }
                    >
                      May 21, 2019
                    </span>
                  </p>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      className="md-grid footer--grid"
    >
      <div
        className="md-cell md-cell--6 md-cell--8-tablet"
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
            }
          }
        >
          <div
            style={
              Object {
                "marginRight": 10,
              }
            }
          >
            <img
              alt="Celestial Loom Logo"
              src="File contents (usually images)"
              style={
                Object {
                  "width": 50,
                }
              }
            />
          </div>
          <h3
            className="footer--title"
          >
            The Celestial Loom
          </h3>
        </div>
        <div
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <p
            className="footer--text"
          >
            Â©
             Copyright 2019, Nikki Davenport.
          </p>
          <p
            className="footer--text"
          >
            Cover photography by Photo by Anastasia Dulgier on Unsplash.
          </p>
          <p
            className="footer--text"
          >
            Built with Gatsby and Netflify CMS.
          </p>
        </div>
      </div>
      <div
        className="md-cell md-cell--6 md-cell--8-tablet"
      >
        <div
          className="md-grid"
        >
          <div
            className="md-cell md-cell--6 md-cell--4-tablet"
          >
            <p
              className="subheader"
            >
              Links
            </p>
            <ul
              className="footer--list"
            >
              <li
                className="footer--line-item"
              >
                <a
                  href="/home/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Home
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/about-nikki/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  About Nikki
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/getting-started/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Getting Started
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/posts/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Posts
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/events/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Events
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/services/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Services
                </a>
              </li>
              <li
                className="footer--line-item"
              >
                <a
                  href="/toolbox/"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Astro Toolbox
                </a>
              </li>
            </ul>
          </div>
          <div
            className="md-cell md-cell--6 md-cell--4-tablet"
          >
            <p
              className="subheader"
            >
              Contact
            </p>
            <ul
              className="footer--list"
            >
              <li>
                <div
                  className="footer--link"
                >
                  <i
                    className="md-icon material-icons"
                    style={
                      Object {
                        "color": "white",
                      }
                    }
                  >
                    email
                  </i>
                  <a
                    className="footer--link__link"
                    href="mailto: nikiastro@att.net"
                  >
                    Email Nikki
                  </a>
                </div>
              </li>
              <li>
                <div
                  className="footer--link"
                >
                  <svg
                    fill="white"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm3 8h-1.35c-.538 0-.65.221-.65.778v1.222h2l-.209 2h-1.791v7h-3v-7h-2v-2h2v-2.308c0-1.769.931-2.692 3.029-2.692h1.971v3z"
                    />
                  </svg>
                  <a
                    className="footer--link__link"
                    href="https://www.facebook.com/celestialloom/"
                  >
                    Facebook
                  </a>
                </div>
              </li>
              <li>
                <div
                  className="footer--link"
                >
                  <svg
                    fill="white"
                    height={20}
                    viewBox="0 0 24 24"
                    width={20}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6.066 9.645c.183 4.04-2.83 8.544-8.164 8.544-1.622 0-3.131-.476-4.402-1.291 1.524.18 3.045-.244 4.252-1.189-1.256-.023-2.317-.854-2.684-1.995.451.086.895.061 1.298-.049-1.381-.278-2.335-1.522-2.304-2.853.388.215.83.344 1.301.359-1.279-.855-1.641-2.544-.889-3.835 1.416 1.738 3.533 2.881 5.92 3.001-.419-1.796.944-3.527 2.799-3.527.825 0 1.572.349 2.096.907.654-.128 1.27-.368 1.824-.697-.215.671-.67 1.233-1.263 1.589.581-.07 1.135-.224 1.649-.453-.384.578-.87 1.084-1.433 1.489z"
                    />
                  </svg>
                  <a
                    className="footer--link__link"
                    href="https://twitter.com/nikiastro"
                  >
                    Twitter
                  </a>
                </div>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
